wsd/ClientRequestDispatcher.cpp

```diff
@@ -926,18 +928,36 @@ void ClientRequestDispatcher::handleIncomingMessage(SocketDisposition& dispositi
         return;
     }
 
-    // If we succeeded - remove the request header from our input buffer
-    socket->eraseFirstInputBytes(map._headerSize);
-    // Some handlers, like handleClientProxyRequest, used in the richdocumentscode
-    // Nextcloud 'Built-in CODE Server' case, launch a task that will consume the
-    // message contents later, while some like the file server, have already handled
-    // the message and we need to discard the message contents (cool.html POST) now
-    // because we expect multiple requests per socket
-    if (served)
+    if( socket->getInBuffer().size() > 0 ) // buffer may have been cleared already due to ignoreInput
     {
-        //remove the request body from our input buffer
-        socket->eraseFirstInputBytes(map._messageSize - map._headerSize);
+        // If we succeeded - remove the request header from our input buffer
+        socket->eraseFirstInputBytes(map._headerSize);
+        // Some handlers, like handleClientProxyRequest, used in the richdocumentscode
```

Comment:
Looks really good - but I'd prefer not to have such big comments in-line in the code - but have them in the unit test and/or git commit log - with perhaps a public github ticket reference here for documentation if necessary in future.

Reply:
will rename served -> servedSync and add the notion to it. This patch clarifies this whole commend AFAIK and hence renders it redundant. Will add a ref to the issues, yup.